### PR TITLE
[Core] Finalize state resolvers

### LINKED ISSUE #7137: [Core][DX] Make private properties and methods in StateResolvers protected
Hello,
The provided StateResolvers like https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Core/StateResolver/OrderPaymentStateResolver.php have only private and public properties and methods. 

This make more difficult to extend them as you almost need to duplicate all code in the classes. And  extending this is probably needed if you add more states in place. 

I suggest to change those private to protected. ¿Thoughts?

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes #7137 |
| License         | MIT |

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Could you add this to UPGRADE file?

**Comment by lchrusciel** [comment]
@pjedrzejewski done

File: UPGRADE.md [hunk 1]
```diff
@@ -25,6 +25,8 @@
   and a scope 'tax' or 'all' is required in `Sylius\Component\Core\OrderProcessing\OrderTaxesProcessor`. 
   A migration file has been prepared  which fill in "all" as scope for zones that didn't have it specified, so they will be resolved by new implementation.
 
+* State resolvers has been made final. In order to change its behavior please use appropriate design pattern or provide your own implementation.
```
Comment (pamil) [inline UPGRADE.md]: has -> have
its -> theirs
use appropriate design pattern -> decorate them / use Decorator pattern

**Comment by pjedrzejewski** [comment]
Thank you Łukasz!

**Pull request merged**